nomt/src/beatree/ops/mod.rs

```diff
@@ -59,24 +60,13 @@ pub fn lookup(
 /// node pointer whose separator is less than or equal to the given key.
 fn search_branch(branch: &branch::BranchNode, key: Key) -> Option<(usize, PageNumber)> {
     let prefix = branch.prefix();
-
-    match key.view_bits::<Msb0>()[..prefix.len()].cmp(prefix) {
-        Ordering::Equal => {}
-        Ordering::Less => return None,
-        Ordering::Greater => {
```

Comment:
actually, the only thing that needs to change here is that this branch needs a match guard `if k == n`. then just have a small branch for the comparisons in the binary search.